### ISSUE TITLE
Perl parser spins indefinitely

### ISSUE THREAD

**Issue by stanhu** [jneen/rouge#539]
From https://gitlab.com/gitlab-org/gitlab-ce/issues/21787:

Specifying this Perl text appears to be make the Lexer spin indefinitely:

``` perl
=head1 hardToParse
  It's going to be hard to parse
=cut

sub substitutionOperator
{
  s!\\!\\\\!g;
  s!\\!\\\\!g;
  s!"!\\"!g;
  s!\(!\\\(!g;
  s!\)!\\\)!g;
}


sub substitutionOperator
{
  s!\\!\\\\!g;
  s!\\!\\\\!g;
  s!"!\\"!g;
  s!\(!\\\(!g;
  s!\)!\\\)!g;
}


sub substitutionOperator
{
  s!\\!\\\\!g;
  s!\\!\\\\!g;
  s!"!\\"!g;
  s!\(!\\\(!g;
  s!\)!\\\)!g;
}
```

**Comment by jneen** [comment]
Okay, I found some slow regexen in the perl lexer, and it seems to be fine as of 63e5bee.
